test/lib/dummy_partitioner.hh

```diff
@@ -37,17 +36,11 @@ public:
     // ordered associative container (std::map) that has dht::token as keys.
     // Values will be ignored.
     template <typename T>
-    dummy_partitioner(const dht::i_partitioner& partitioner, const std::map<dht::token, T>& something_by_token)
-        : i_partitioner(smp::count)
-        , _partitioner(partitioner)
+    dummy_sharding_info(const dht::sharding_info& sharding, const std::map<dht::token, T>& something_by_token)
+        : sharding_info(sharding)
         , _tokens(boost::copy_range<std::vector<dht::token>>(something_by_token | boost::adaptors::map_keys)) {
     }
 
-    virtual dht::token get_token(const schema& s, partition_key_view key) const override { return _partitioner.get_token(s, key); }
-    virtual dht::token get_token(const sstables::key_view& key) const override { return _partitioner.get_token(key); }
-    virtual bool preserves_order() const override { return _partitioner.preserves_order(); }
-    virtual const sstring name() const override { return _partitioner.name(); }
     virtual unsigned shard_of(const dht::token& t) const override;
```

Comment:
Ok. I see that sharding_info::shard_of() preserves the virtualness of i_partitioner::shard_of() and does not introduce it. So it's okay to keep it. though we should try to remove it later. It certainly doesn't seem to be easy.